client/src/client.rs

```diff
@@ -184,8 +219,42 @@ impl<ST: StorageType, SSA: SignalServerAPI> Client<ST, SSA> {
         ))
     }
 
-    pub async fn login() -> Result<Self> {
-        todo!("Implement when Storage<Device> is done")
+    pub async fn login(
+        database_url: &str,
+        cert_path: &str,
+        server_url: &str,
+    ) -> Result<Client<Device, SignalServer>> {
+        let pool = Client::<T, SSA>::connect(database_url, false).await?;
+        let device = Device::new(pool.clone());
+        let contacts = match device.load_contacts().await {
+            Ok(contacts) => {
+                let mut c = HashMap::new();
+                for contact in contacts {
+                    c.insert(contact.service_id, contact);
+                }
+                Ok(c)
+            }
+            Err(err) => Err(SignalClientError::DatabaseError(format!("{err}"))),
+        }?;
+        let (one_time, signed, kyber) = device
+            .get_key_ids()
+            .await
+            .map_err(|err| SignalClientError::DatabaseError(format!("{err}")))?;
```

Comment:
lav en struct
```
#[derive(Debug, Display, Error)]
pub struct DatabaseError(SignalProtocolError)
```
or lav SignalClientError om til
```
SignalClientError::DatabaseError(DatabaseError)
```
så kan du måske skrive
```suggestion
            .map_err(DatabaseError)?;
```